app/controllers/application_controller.rb

```diff
@@ -7,11 +7,38 @@ class ApplicationController < ActionController::Base
 
   private
   @@name_path = nil
-  
+   
   def logged_in?
     @@name_path = request.env['PATH_INFO']
     redirect_to login_path and return unless
       (@current_user = User.find_by_id(session[:user_id])).kind_of?(User)
   end
   
+  def auth_user?
+    if User.find_by_id(session[:user_id]).type_user == "Student"
+      redirect_path
+    end 
+  end 
+  
+  def redirect_path
+    if @@name_path.first(5) == "/apps" 
+      redirect_helper(apps_path, 'Error: Only Staff can create, edit and destroy apps and engagements')
+    elsif @@name_path.first(5) == "/orgs"
+      redirect_helper(orgs_path, 'Error: Only Staff can create, edit and destroy orgs')
+    elsif @@name_path.first(5) == "/user"
+      redirect_helper(users_path, 'Error: Only Staff can create and edit users')
+    elsif @@name_path.first(5) == "/crea"
+      redirect_helper(apps_path, 'Error: Only Staff can create apps, orgs, users')
+    elsif @@name_path.first(5) == "/enga"
+      parts = @@name_path.split("/")
+      path = "/" + parts[1] + "/" + parts[2] + "/" + parts[3]
+      redirect_helper(path, 'Error: Only Staff can create, edit and destroy apps, engagements and iterations') 
```

Comment:
Should probably get DRY'd out during a future iteration, but this is definitely a complicated method to implement so good job getting it done